state/aws.go

```diff
@@ -30,12 +31,22 @@ type AWS struct {
 func NewAWS(c *config.Config) AWS {
 	sess := session.Must(session.NewSession())
 
+	awsConfig := &aws_sdk.Config{}
+
+	if len(c.AWS.APPRoleArn) > 0 {
+		log.Debugf("Using %s role", c.AWS.APPRoleArn)
+		creds := stscreds.NewCredentials(sess, c.AWS.APPRoleArn)
+		awsConfig = &aws_sdk.Config{Credentials: creds}
+	} else {
+		log.Info("There is no role specified.")
```

Comment:
I'm not sure this `else` bit is necessary.

Reply:
you're right :)